health/health.c

```diff
@@ -1,5 +1,7 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 
+#include <assert.h>
```

Comment:
We don't really use assert anywhere (the one in the freeipmi plugin is an exception). Can't you handle the cases with unlikely and a message + correction if possible?

Reply:
OK, I'll do that. Assertions can be handful since it stops the engine gracefully when a never-happening condition is met. By the way, I'll remove them.

Reply:
I removed the asserts. There a bunch of TODOs which I address in this PR.